10월 18, 2014 7:22:45 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter <init>
정보: Setting Up identifiers & ports for the network service to listen on
10월 18, 2014 7:22:45 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$StartHandler onNext
정보: StartTime: com.microsoft.wake.time.event.StartTime[1413627765960]
10월 18, 2014 7:22:46 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$AllocatedEvaluatorHandler onNext
정보: Received an AllocatedEvaluator. Submitting it.
10월 18, 2014 7:22:46 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 1, Outstanding requests: 1
10월 18, 2014 7:22:46 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 2, Outstanding requests: 1
10월 18, 2014 7:22:46 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$AllocatedEvaluatorHandler onNext
정보: Received an AllocatedEvaluator. Submitting it.
10월 18, 2014 7:22:46 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 3, Outstanding requests: 0
10월 18, 2014 7:22:46 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 3, Outstanding requests: 0
10월 18, 2014 7:22:46 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$AllocatedEvaluatorHandler onNext
정보: Received an AllocatedEvaluator. Submitting it.
10월 18, 2014 7:22:46 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-2-1413627766089"
state: RUNNING
 
10월 18, 2014 7:22:46 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-3-1413627766172"
state: RUNNING
 
10월 18, 2014 7:22:46 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-1-1413627765970"
state: RUNNING
 
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$ActiveContextHandler onNext
정보: Received a RunningEvaluator with ID: SGD_Linear
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$ActiveContextHandler onNext
정보: Received a RunningEvaluator with ID: SGD_Linear
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$ActiveContextHandler onNext
정보: Received a RunningEvaluator with ID: SGD_Linear
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: All context are running
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Setting Up Structures for creating Group Comm Operator Configurations
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Registering ComputeTask1 with jw-PC:7001
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Registering ComputeTask2 with jw-PC:7002
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.TaskSubmitter onNext
정보: Creating Operator Configs
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$RunningTaskHandler onNext
정보: Task "ComputeTask1" is running!
10월 18, 2014 7:22:49 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$RunningTaskHandler onNext
정보: Task "ComputeTask2" is running!
10월 18, 2014 7:22:50 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$RunningTaskHandler onNext
정보: Task "ControllerTask" is running!
10월 18, 2014 7:22:50 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Task ComputeTask1 is done.
10월 18, 2014 7:22:50 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Task ComputeTask2 is done.
10월 18, 2014 7:22:50 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Releasing Context SGD_Linear.
10월 18, 2014 7:22:50 오후 snu.BDCS.SGD_Linear_Regression.LinearSGD_Driver$CompletedTaskHandler onNext
정보: Releasing Context SGD_Linear.
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onError
경고: Shutting down the Driver with an exception: 
java.lang.RuntimeException: No handler bound for FailedTask: com.microsoft.reef.driver.task.FailedTask id=ControllerTask failed: Task.call() threw an Exception.
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultTaskFailureHandler.onNext(DefaultTaskFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultTaskFailureHandler.onNext(DefaultTaskFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.runtime.common.evaluator.task.exceptions.TaskCallFailure: Task.call() threw an Exception.
	at com.microsoft.reef.runtime.common.evaluator.task.TaskRuntime.runTask(TaskRuntime.java:267)
	at com.microsoft.reef.runtime.common.evaluator.task.TaskRuntime.run(TaskRuntime.java:123)
	... 1 more
Caused by: java.lang.ArrayIndexOutOfBoundsException: 6
	at com.microsoft.reef.examples.groupcomm.matmul.DenseVector.get(DenseVector.java:42)
	at snu.BDCS.SGD_Linear_Regression.VectorConcat.apply(VectorConcat.java:36)
	at snu.BDCS.SGD_Linear_Regression.VectorConcat.apply(VectorConcat.java:20)
	at com.microsoft.reef.io.network.group.impl.operators.basic.ReduceOp$Receiver.reduce(ReduceOp.java:118)
	at com.microsoft.reef.io.network.group.impl.operators.basic.ReduceOp$Receiver.reduce(ReduceOp.java:106)
	at snu.BDCS.SGD_Linear_Regression.ControllerTask.call(ControllerTask.java:94)
	at com.microsoft.reef.runtime.common.evaluator.task.TaskRuntime.runTask(TaskRuntime.java:262)
	... 2 more

10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Dirty shutdown with outstanding containers.
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Force shutdown of: ProcessContainer{containedID='Node-2-1413627766089', nodeID='Node-2', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-2-1413627766089}
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-2-1413627766089', nodeID='Node-2', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-2-1413627766089}
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0xcc6b5c43, /169.254.245.8:1307 => /169.254.245.8:17097] local: /169.254.245.8:17097 remote: /169.254.245.8:1307
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:17097 remote: /169.254.245.8:1307 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-2-1413627766089"
state: FAILED
exit_code: 1
 
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0x33ef873e, /169.254.245.8:1573 => /169.254.245.8:5678] local: /169.254.245.8:5678 remote: /169.254.245.8:1573
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager close
경고: Force resource release because the client closed the clock.
java.lang.IllegalStateException: Scheduling alarm on a closed clock
	at com.microsoft.wake.time.runtime.RuntimeClock.scheduleAlarm(RuntimeClock.java:78)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.close(EvaluatorManager.java:174)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onEvaluatorDone(EvaluatorManager.java:346)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onEvaluatorStatusMessage(EvaluatorManager.java:323)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onEvaluatorHeartbeatMessage(EvaluatorManager.java:299)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorHeartbeatHandler.onNext(EvaluatorHeartbeatHandler.java:58)
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorHeartbeatHandler.onNext(EvaluatorHeartbeatHandler.java:33)
	at com.microsoft.wake.remote.impl.HandlerContainer.onNext(HandlerContainer.java:143)
	at com.microsoft.wake.remote.impl.HandlerContainer.onNext(HandlerContainer.java:34)
	at com.microsoft.wake.remote.impl.OrderedPullEventHandler.onNext(OrderedRemoteReceiverStage.java:167)
	at com.microsoft.wake.remote.impl.OrderedPullEventHandler.onNext(OrderedRemoteReceiverStage.java:149)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorCompletionHandler onNext
정보: Received CompletedEvaluator: CompletedEvaluator{id='Node-3-1413627766172'}
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:5678 remote: /169.254.245.8:1573 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Force shutdown of: ProcessContainer{containedID='Node-1-1413627765970', nodeID='Node-1', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-1-1413627765970}
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-1-1413627765970', nodeID='Node-1', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-1-1413627765970}
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0xd63cafb6, /169.254.245.8:1304 => /169.254.245.8:17097] local: /169.254.245.8:17097 remote: /169.254.245.8:1304
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:17097 remote: /169.254.245.8:1304 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0x8629aaf7, /169.254.245.8:1706 => /169.254.245.8:5678] local: /169.254.245.8:5678 remote: /169.254.245.8:1706
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:5678 remote: /169.254.245.8:1706 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager onEvaluatorException
경고: Failed evaluator: Node-2-1413627766089
com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-2-1413627766089] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-1-1413627765970"
state: FAILED
exit_code: 1
 
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onError
경고: Shutting down the Driver with an exception: 
java.lang.RuntimeException: No handler bound for FailedEvaluator: FailedEvaluator{id='Node-2-1413627766089'}
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-2-1413627766089] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	... 1 more

10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager setStatus
경고: Illegal state transiton: 'FAILING'->'FAILING'
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.driver.ContainerManager close
경고: Force shutdown of: ProcessContainer{containedID='Node-3-1413627766172', nodeID='Node-3', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-3-1413627766172}
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-3-1413627766172', nodeID='Node-3', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-3-1413627766172}
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
정보: Unexpected exception from downstream. channel: [id: 0x297e7771, /169.254.245.8:1572 => /169.254.245.8:5678] local: /169.254.245.8:5678 remote: /169.254.245.8:1572
10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.NettyChannelHandler exceptionCaught
경고: Unexpected exception from downstream.
java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.UnpooledUnsafeDirectByteBuf.setBytes(UnpooledUnsafeDirectByteBuf.java:446)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:879)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:225)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:114)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:22:50 오후 com.microsoft.wake.remote.transport.netty.AbstractNettyEventListener exceptionCaught
경고: ExceptionEvent: local: /169.254.245.8:5678 remote: /169.254.245.8:1572 :: java.io.IOException: 현재 연결은 원격 호스트에 의해 강제로 끊겼습니다
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager onEvaluatorException
경고: Failed evaluator: Node-1-1413627765970
com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-1-1413627765970] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	at java.lang.Thread.run(Thread.java:745)

10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver onResourceStatus
정보: Sending resource status: identifier: "Node-3-1413627766172"
state: FAILED
exit_code: 1
 
10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onError
경고: Shutting down the Driver with an exception: 
java.lang.RuntimeException: No handler bound for FailedEvaluator: FailedEvaluator{id='Node-1-1413627765970'}
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-1-1413627765970] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	... 1 more

10월 18, 2014 7:22:50 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager setStatus
경고: Illegal state transiton: 'FAILING'->'FAILING'
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ContainerManager release
정보: Releasing Container with containerId [ProcessContainer{containedID='Node-1-1413627765970', nodeID='Node-1', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-1-1413627765970}]
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.evaluator.Evaluators close
경고: Unclean shutdown of evaluator Node-2-1413627766089
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-1-1413627765970', nodeID='Node-1', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-1-1413627765970}
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.evaluator.Evaluators close
경고: Unclean shutdown of evaluator Node-1-1413627765970
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.evaluator.Evaluators close
경고: Unclean shutdown of evaluator Node-3-1413627766172
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager sendJobEndingMessageToClient
경고: Sending message about an unclean driver shutdown.
java.lang.RuntimeException: No handler bound for FailedEvaluator: FailedEvaluator{id='Node-1-1413627765970'}
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:34)
	at com.microsoft.reef.runtime.common.driver.defaults.DefaultEvaluatorFailureHandler.onNext(DefaultEvaluatorFailureHandler.java:26)
	at com.microsoft.reef.runtime.common.utils.BroadCastEventHandler.onNext(BroadCastEventHandler.java:34)
	at com.microsoft.reef.util.ExceptionHandlingEventHandler.onNext(ExceptionHandlingEventHandler.java:42)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:82)
	at com.microsoft.reef.runtime.common.utils.DispatchingEStage$1.onNext(DispatchingEStage.java:79)
	at com.microsoft.wake.impl.ThreadPoolStage$1.run(ThreadPoolStage.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.microsoft.reef.exception.EvaluatorException: Evaluator [Node-1-1413627765970] is assumed to be in state [RUNNING]. But the resource manager reports it to be in state [FAILED]. This means that the Evaluator failed but wasn't able to send an error message back to the driver.
	at com.microsoft.reef.runtime.common.driver.evaluator.EvaluatorManager.onResourceStatusMessage(EvaluatorManager.java:495)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:57)
	at com.microsoft.reef.runtime.common.driver.resourcemanager.ResourceStatusHandler.onNext(ResourceStatusHandler.java:34)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onResourceStatus(ReefRunnableProcessObserver.java:119)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onUncleanExit(ReefRunnableProcessObserver.java:99)
	at com.microsoft.reef.runtime.local.process.ReefRunnableProcessObserver.onProcessExit(ReefRunnableProcessObserver.java:72)
	at com.microsoft.reef.runtime.local.process.RunnableProcess.run(RunnableProcess.java:159)
	... 1 more

10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 2, Outstanding requests: 0
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ContainerManager release
정보: Releasing Container with containerId [ProcessContainer{containedID='Node-2-1413627766089', nodeID='Node-2', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-2-1413627766089}]
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-2-1413627766089', nodeID='Node-2', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-2-1413627766089}
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 1, Outstanding requests: 0
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ContainerManager release
정보: Releasing Container with containerId [ProcessContainer{containedID='Node-3-1413627766172', nodeID='Node-3', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-3-1413627766172}]
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ProcessContainer close
경고: Force-closing a container that is still running: ProcessContainer{containedID='Node-3-1413627766172', nodeID='Node-3', errorHandlerRID='socket://169.254.245.8:17097', folder=C:\Users\jw\Dropbox\git\SGD-LinearRegression_on_REEF\REEF_LOCAL_RUNTIME\SGD_Linear-1413627763949\Node-3-1413627766172}
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.process.RunnableProcess cancel
경고: The child process survived Process.destroy()
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 0, Outstanding requests: 0
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.idle.DriverIdleManager onPotentiallyIdle
정보: All components indicated idle. Initiating Driver shutdown.
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onComplete
정보: Clean shutdown of the Driver.
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager setStatus
경고: Illegal state transiton: 'FAILING'->'SHUTTING_DOWN'
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ContainerManager release
정보: Ignoring release request for unknown containerID [Node-3-1413627766172]
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.idle.DriverIdleManager onPotentiallyIdle
정보: All components indicated idle. Initiating Driver shutdown.
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.local.driver.ResourceManager sendRuntimeStatus
정보: Allocated: 0, Outstanding requests: 0
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onComplete
정보: Clean shutdown of the Driver.
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager setStatus
경고: Illegal state transiton: 'FAILING'->'SHUTTING_DOWN'
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.idle.DriverIdleManager onPotentiallyIdle
정보: All components indicated idle. Initiating Driver shutdown.
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.DriverRuntimeStopHandler onNext
정보: Driver shutdown complete
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager onComplete
정보: Clean shutdown of the Driver.
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.driver.DriverStatusManager setStatus
경고: Illegal state transiton: 'FAILING'->'SHUTTING_DOWN'
10월 18, 2014 7:22:51 오후 com.microsoft.reef.runtime.common.Launcher main
정보: Exiting Launcher.main()
